modules: Kconfig.microchip: Introduce SAM kconfig symbol

ArunMCHP opened the pull request:
Add a new SAM Kconfig symbol and selected it in the
SAM D5x/E5x family devices to enable proper grouping.

This improves configuration consistency for Microchip SAM-based SoCs.

Associated hal_microhcip PR: https://github.com/zephyrproject-rtos/hal_microchip/pull/43

ArunMCHP committed:
drivers: pinctrl: microchip: Fix header include style

Fixed the include directive for mchp_pinctrl_pinmux_sam.h
by replacing quotes with angle brackets.

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

ArunMCHP committed:
modules: Reorganize directory structure and update Kconfig symbols

- Update the path of pinctrl.h in sama7g54_ek.dts
- Select CONFIG_MICROCHIP_SAM for sama7g5 family devices
- Add SAM group Kconfig symbol for proper family grouping
- Rename PIC32C Kconfig symbol to MICROCHIP_PIC32C
  and update references
- Update west.yml for hal-microchip re-organization

Signed-off-by: Arunprasath P <arunprasath.p@microchip.com>

nordicjm commented on soc/microchip/sam/sam_d5x_e5x/Kconfig:
```diff
@@ -3,6 +3,7 @@
 
 config SOC_FAMILY_MICROCHIP_SAM_D5X_E5X
```
there already is this soc family Kconfig, with sam in it, it is unique, it is a soc Kconfig, it's not going to conflict with anything else. SAM is completely generic and if this one symbol selects it I don't see the reason for it, check for `SOC_FAMILY_MICROCHIP_SAM_D5X_E5X` instead

nordicjm replied:
The layout of folders is however you want to do it, that is unrelated to the Kconfigs

nordicjm replied:
If their hal is built for zephyr, it can use the zephyr Kconfig symbols, or do some prefix so it to so it's unique e.g. `MICROCHIP_SAM` because `SAM` is far too common of a word to just block, that means no subsystem in future called SAM can be added because a vendor has decided to pick a name. SOC_* Kconfigs are first come first serve for SoCs, but that is not the same for non-SoC Kconfigs for SoCs

nandojve replied:
@ArunMCHP , @NhMchp ,

Could you add **MICROCHIP_** prefix on your symbols ?

```diff
-config PIC32C
+config MICROHIP_PIC32C
	bool

-config SAM
+config MICROHIP_SAM
	bool
```

I'm not sure but I think, depending of your goal, switch from bool to string could bring advantages:

```
config MICROHIP_SAM
	string
	default "sam"
```

ArunMCHP replied:
Hi @nordicjm, @nandojve,

Thank you for your feedback and suggestions.
I have now updated the Kconfig symbols with `MICROCHIP_` prefix for both `PIC32C` and `SAM`, as recommended.

Kindly review and approve the PR, if there are no other comments.

nandojve commented on soc/microchip/pic32c/pic32cm_jh/Kconfig:
```diff
@@ -3,7 +3,7 @@
 config SOC_FAMILY_MICROCHIP_PIC32CM_JH
 	select ARM
-	select PIC32C
+	select MICROCHIP_PIC32C
```
This change should be atomic together with the `modules/Kconfig.microchip` because it breaks the build.
My suggestion is that you create a commit after the west changes (because you may need to adapt the hal - not sure) with one commit renaming the PIC32C -> MICROCHIP_PIC32C. Then later, you do the normal flow.
This avoids ping pong commits and the rename to PIC32C -> MICROCHIP_PIC32C can be undestood as a preparation to the next commits that follows.

ArunMCHP commented:
Hi @nandojve, @scottwcpg, @nordicjm,

Thanks for reviewing and approving the PR. 

I’ve now updated the `west.yml` with the new revision ID, rebased the branch, and updated the PR. Could you kindly approve it again?
